packages/react/select/src/Select.tsx

```diff
@@ -1219,7 +1219,7 @@ const BubbleSelect = React.forwardRef<HTMLSelectElement, React.ComponentPropsWit
      */
     return (
       <VisuallyHidden asChild>
-        <select {...selectProps} ref={composedRefs} defaultValue={value} />
+        <select {...selectProps} ref={composedRefs} value={value} defaultValue={value} />
```

Comment:
The `defaultValue` should be fine but the `value` will break form `onChange` event. See the comment above:

>      * We purposefully do not add the `value` attribute here to allow the value
>      * to be set programatically and bubble to any parent form `onChange` event.
>      * Adding the `value` will cause React to consider the programatic
>      * dispatch a duplicate and it will get swallowed.

Reply:
I'm blind, sorry!
So, what would be the best way to fix this ? 
https://github.com/radix-ui/primitives/issues/1223

Reply:
I think the `defaultValue` is all that's needed

Reply:
perhaps something like `<select key={children} ....>` would solve it?

Reply:
That's a cool idea, but it doesn't work because there are no `children` as they are portalled in.
Even if trying with an arbitrary key update at any moment in time, it doesn't seem to work either, guessing because the children are not in there rendered.

It seems to me that the native select will only work with the `defaultValue` if they are rendered at the same time, for example this doesn't work either:

https://codesandbox.io/s/serverless-architecture-e77e62?file=/src/App.js

Reply:
scrap that, `MutationObserver` could work maybe? observe `childList` and change `key`.

Reply:
seems the only way would be to render `BubbleSelect` in viewport so children can be passed directly to it and provide some context that `item` can use to determine what to render based on whether it's native select or not 😬

Reply:
I don't think that would work as you still need both the custom and native select. Not sure how you would otherwise redirect the native `option` inside the native `select`, unless I'm not understanding what you're suggesting.

Reply:
suggesting something like this:

```jsx
const Viewport = () => {
  return (
    <Collection.Slot>
      <Primitive.div>{children}</Primitive.div>

      <IsInsideNativeSelect.Provider value={true}>
        <BubbleInput>{children}</BubbleInput>
      </IsInsideNativeSelect.Provider>
    </Collection.Slot>
  );
};

const Item = () => {
  const isInsideNativeSelect = React.useContext(IsInsideNativeSelect)
  return isInsideNativeSelect ? <option /> : <Item />
};

```

Reply:
Oh I see, duplicating the children basically. hmmm, man that seems rather heavy 😀

The React tree would now be double the size, and also not sure about the `select` being inside the viewport, although we could potentially portal the whole subtree higher up maybe?

Reply:
how has that doubled in size? it's rendering the same number of `options` as we are currently. it's doing the exact same thing but without portalling.

Reply:
> how has that doubled in size? it's rendering the same number of options as it is currently. it's doing the same thing but without portalling.

In terms of DOM yes, but not in terms of React tree, because you now get double amounts of `SelectItem`s in the tree with half of them rendering to different elements in DOM tree.

> i don't think it really matters where the select renders given that it's hidden? oh but do you mean flex styling n stuff is impacted perhaps? hmm, is it if it's hidden?

I was more referring to the fact that if it was in the viewport, it would be in the "collapsed" part. I know we do render it in a document fragment for typeahead, but I'm not sure if everything would work in a fragment if we need more control, like focus events, etc

Worth a shot anyway.